src/ir.cpp

```diff
@@ -19707,17 +19707,25 @@ static IrInstGen *ir_analyze_fn_call(IrAnalyze *ira, IrInst* source_instr,
     }
 
     if (modifier == CallModifierCompileTime) {
+        bool extern_fn_in_typeof = false;
+
         // No special handling is needed for compile time evaluation of generic functions.
         if (!fn_entry || fn_entry->body_node == nullptr) {
-            ir_add_error(ira, &fn_ref->base, buf_sprintf("unable to evaluate constant expression"));
-            return ira->codegen->invalid_inst_gen;
+            // We keep evaluating extern functions directly in TypeOfs
+            if (fn_entry && source_instr->scope->id == ScopeIdTypeOf) {
```

Comment:
You can heavily simplify this by using something like this:
```C
if (fn_entry && source_instr->scope->id == ScopeIdTypeOf) {
    AstNode *return_type_node = fn_proto_node->data.fn_proto.return_type;
    ZigType *specified_return_type = ir_analyze_type_expr(ira, source_instr->scope, return_type_node);
    return ir_const_undef(ira, source_instr, specified_return_type);
}
```

What's the definition of "extern" you're using here?

Reply:
I'm checking for an extern function prototype, I will use the AST node instead for more clarity.  
Updating soon.